typescript/vrotsc/src/compiler/transformer/fileTransformers/workflow/decorators/helpers/graph.ts

```diff
@@ -14,335 +14,339 @@
+			treeNode.branches = null;
+		}
+		else if (treeNode.name === Graph.START) { // start node is on same row as first branch
+			treeNode.row = this.getNode(treeNode.branches[0]).row;
+		}
+		else { // branch has children - put on row approximately between the first and last child's;
+			treeNode.row = Math.floor((this.getNode(treeNode.branches[0]).row + this.getNode(treeNode.branches[treeNode.branches.length - 1]).row) / 2);
+			// in case of a node on the same row back referencing this one - avoid arrows crossing intermediate node(s)
+			while (
+				treeNode.sources.find((src, ind) => !!ind && (src in this.visitedNodes)  // visited, but not parent
+					&& this.getNode(src).row === treeNode.row && this.getNode(src).col > treeNode.col + 1) // further than next col on same row
+			) {
+				treeNode.row++;
 			}
 		}
-		return false;
+		this.gridTracker.lastRowIndPerCol[treeNode.col] = treeNode.row;
+		this.gridTracker.totalRows = Math.max(this.gridTracker.totalRows, 1 + treeNode.row);
 	}
 
 	/**
-	 * Builds the tree
-	 *
-	 * Calculates the x and y positions of each leaf
-	 *
-	 * @param {GraphNode} origin - The node to start building the tree from
-	 * @param {Set<GraphNode>} visited - The set of visited nodes
-	 * @param {number} x - The x position of the new node
-	 * @param {number} y - The y position of the new node
+	 * Finds a node with the given name. Checks visited nodes first.
+	 * @param {string} name
+	 * @returns {GraphNode}
+	 * @throws Error if the Graph contains no node with the given name.
 	 */
-	private buildTree(
-		origin: GraphNode,
-		visited: Set<GraphNode> = new Set<GraphNode>(),
-		x: number = this.nodeSpacing,
-		y: number = this.height / 2
-	) {
-		while (this.hasOverlap(x, y)) {
-			y += this.nodeSpacing;
-		}
-
-		origin.x = x;
-		origin.y = y;
-
-		let positions = {
-			x: x + this.nodeSpacing,
-			y: y
-		};
-
-		if (origin.targets.length > 1) {
-			positions.y -= this.nodeSpacing * (origin.targets.length - 1) / 2;
+	public getNode(name: string): GraphNode {
+		const node = this.visitedNodes[name] || this.nodes.find(node => node.name === name);
+		if (!node) {
+			throw new Error(`Node "${name}" not found`);
 		}
 
-		origin.targets.forEach(targetName => {
-			const target = this.getNode(targetName);
-
-			if (!this.tree.leaves[target.name]) {
-				this.tree.leaves[target.name] = {
-					node: target,
-					leaves: []
-				};
-			}
-
-			const targetLeaf = this.getLeaf(target.name);
-			this.getLeaf(origin.name).leaves.push(targetLeaf);
-
-			// LPS (`Loop Prevention System`)
-			if (visited.has(target)) return;
-			visited.add(target);
-
-			this.buildTree(targetLeaf.node, visited, positions.x, positions.y);
-			positions.y += this.nodeSpacing;
-		});
+		return node;
 	}
 
 	/**
-	 * Calculates all the positions of the tree leaves
+	 * Use to draw a graph in the console
 	 *
-	 * If this method is called multiple times, it will only calculate the positions once
+	 * @WARN: Used for debugging purposes
```

Comment:
Well it's getting drawn every time. We don't need to output this information to the user.

Reply:
IMO seeing the rough WF schema before pushing to the environment does bring some value. If you disagree, I can remove the draw() method altogether.

Reply:
We are writing a transpiler. It either works, or it returns 1 and fails. Any error needs to fail immediately and we need to inform the user where the error was to the best of our abilities

"Warnings" are also applicable. You can add warnings if we managed to somehow gracefully move on.

This all being said, look up where you have `console.*` and see if you really need it